integration-tests/tests/src/tests/sync/realm-conversions.ts

```diff
@@ -0,0 +1,307 @@
+    });
+
+    interface ConversionTestConfig {
+      encrypted: boolean;
+      sync: boolean;
+    }
+
+    interface ConversionTestContext {
+      configSrc: Realm.Configuration;
+      configDst: Realm.Configuration;
+    }
+
+    /** Returns a Realm Config from a given conversion test configuration. */
+    async function getRealmConfig(
+      context: AppContext,
+      testConfig: ConversionTestConfig,
+      isDestination?: boolean,
+    ): Promise<Realm.Configuration> {
+      let sync;
+      let encryptionKey;
+      if (testConfig.sync) {
+        const credentials = await getRegisteredEmailPassCredentials(context.app);
+        sync = {
+          user: await context.app.logIn(credentials),
+          partitionValue: "foo",
+          _sessionStopPolicy: "immediately" as Realm.SessionStopPolicy, // Make it safe to delete files after realm.close()
+        };
+      }
+
+      if (testConfig.encrypted) {
+        encryptionKey = testConfig.sync ? context.encryptedKeySync : context.encryptedKeyLocal;
+      }
+
+      return {
+        sync,
+        encryptionKey,
+        schema: [DogForSyncSchema],
+        path: `dog${testConfig.sync ? "_sync" : "_local"}${testConfig.encrypted ? "_encrypted" : ""}${
+          isDestination ? "_dst" : ""
+        }.realm`,
+      };
+    }
+
+    async function itShouldConvertRealm(source: ConversionTestConfig, destination: ConversionTestConfig) {
+      const testConfigAsString = (testConfig: ConversionTestConfig) =>
+        `${testConfig.sync ? "synced" : "local"}, ${testConfig.encrypted ? "encrypted" : "unencrypted"}`;
+
+      it(`convert ${testConfigAsString(source)} -> ${testConfigAsString(
+        destination,
+      )}`, async function (this: AppContext & ConversionTestContext) {
+        this.configSrc = await getRealmConfig(this, source);
+        this.configDst = await getRealmConfig(this, destination, true);
+
+        const realmSrc = await Realm.open(this.configSrc);
+
+        expect(realmSrc.objects("Dog").length).equals(0);
+        realmSrc.write(() => {
+          for (let i = 0; i < 5; i++) {
+            realmSrc.create("Dog", {
+              _id: new BSON.ObjectId(),
+              breed: "Domestic Short Hair",
+              name: `Dog no. ${i}`,
+              partition: "foo",
+            });
+          }
+        });
+
+        if (source.sync) {
+          await realmSrc.syncSession?.uploadAllLocalChanges();
+          await realmSrc.syncSession?.downloadAllServerChanges();
+        }
+        realmSrc.writeCopyTo(this.configDst);
+        realmSrc.close();
+
+        const realmDst = await Realm.open(this.configDst);
+        if (destination.sync) {
+          await realmDst.syncSession?.downloadAllServerChanges();
+        }
+        expect(realmDst.objects("Dog").length).equals(5);
+        realmDst.close();
+      });
+    }
+
+    afterEach(async function (this: AppContext & ConversionTestContext) {
+      const configSrc = this.currentTest?.ctx?.configSrc;
```

Comment:
Not sure if there is a better way to retreive this information / accomplish... this seems like a rather ugly way to do so... but it does work.